homework1/task1/version2.py

```diff
@@ -0,0 +1,22 @@
+# make list flatten
+# lst = [1, [2, 3], 4, [[6, 7]]]
+# to
+# lst = [1, 2, 3, 4, 6, 7]
+
+# solution from acquaintances
+
+input_list = [1, [2, 3], 4, [[6, 7]]]
+
+
+def flatten_array(input_lst):
+    index = 0
+    while index < len(input_lst):
+        if type(input_lst[index]) != list:
+            index += 1
+            continue
+        input_lst = input_lst[0:index] + input_lst[index] + input_lst[index + 1:]
```

Comment:
it's very bad practice to modify value of input variable. When you execute the function, your expectation is to receive a result from modification of the input parameter - you would expect having the original input param with the same value as before the call.

Reply:
but yes, the algorithms works fine and solves the task :)

Reply:
Thank you. Code is changed